spec/ruby/fixtures/code/concurrent_require_fixture.rb

```diff
@@ -0,0 +1,2 @@
+
+Thread.new { ScratchPad.recorded.require(__FILE__) }.join(0.1)
```

Comment:
Why the `0.1` and not just `.join`?

Reply:
Because that would cause a deadlock. The `ThreadShield` is help by the main thread, so that subthread will lock on the `require` call until the mainthread is done requiring.

Reply:
Ah yes, right.
We need to fix that spec then, because it will leak a Thread. We should `join` in the caller to make sure that Thread doesn't live longer than the spec example.
You can use `CHECK_LEAKS=true` to detect such resources leaking between spec examples.

Also this will then wait 100ms just to ensure the thread starts, that's very slow for a ruby/spec spec, it's a lot more efficient to rely on `Thread#stop?` (or status) instead as other specs do.

Reply:
Hum, good point, I'll improve that tomorrow (sorry for not waiting on your second review)